openlibrary/catalog/add_book/match.py

```diff
@@ -53,13 +47,434 @@ def editions_match(candidate, existing):
+        all isbns in "isbn": []
+        authors have db_name (name with dates)
+    """
+    rec['full_title'] = rec['title']
+    if subtitle := rec.get('subtitle'):
+        rec['full_title'] += ' ' + subtitle
+    expanded_rec = build_titles(rec['full_title'])
+    expanded_rec['isbn'] = []
+    for f in 'isbn', 'isbn_10', 'isbn_13':
+        expanded_rec['isbn'].extend(rec.get(f, []))
+    if 'publish_country' in rec and rec['publish_country'] not in (
+        '   ',
+        '|||',
+    ):
+        expanded_rec['publish_country'] = rec['publish_country']
+    for f in (
+        'lccn',
+        'publishers',
+        'publish_date',
+        'number_of_pages',
+        'authors',
+        'contribs',
+    ):
+        if f in rec:
+            expanded_rec[f] = rec[f]
+    add_db_name(expanded_rec)
+    return expanded_rec
+
+
+def build_titles(title: str):
+    """
+    Uses a full title to create normalized and short title versions.
+    Used for expanding a set of title variants for matching,
+    not for storing on records or display.
+
+    :param str title: Full title of an edition
+    :rtype: dict
+    :return: An expanded set of title variations
+    """
+    normalized_title = normalize(title)
+    titles = [title, normalized_title, strip_articles(normalized_title)]
+    if m := re_amazon_title_paren.match(normalized_title):
+        titles.append(m.group(1))
+        titles.append(strip_articles(m.group(1)))
+
+    return {
+        'full_title': title,
+        'normalized_title': normalized_title,
+        'titles': list(set(titles)),
+        'short_title': normalized_title[:25],
+    }
+
+
+def within(a, b, distance):
+    return abs(a - b) <= distance
+
+
+def compare_country(e1: dict, e2: dict):
+    field = 'publish_country'
+    if field not in e1 or field not in e2:
+        return (field, 'value missing', 0)
+    if e1[field] == e2[field]:
+        return (field, 'match', 40)
+    # West Berlin (wb) == Germany (gw)
+    if e1[field] in ('gw ', 'wb ') and e2[field] in ('gw ', 'wb '):
+        return (field, 'match', 40)
+    return (field, 'mismatch', -205)
+
+
+def compare_lccn(e1: dict, e2: dict):
+    field = 'lccn'
+    if field not in e1 or field not in e2:
+        return (field, 'value missing', 0)
+    if e1[field] == e2[field]:
+        return (field, 'match', 200)
+    return (field, 'mismatch', -320)
+
+
+def compare_date(e1: dict, e2: dict):
+    if 'publish_date' not in e1 or 'publish_date' not in e2:
+        return ('date', 'value missing', 0)
+    if e1['publish_date'] == e2['publish_date']:
+        return ('date', 'exact match', 200)
+    try:
+        e1_pub = int(e1['publish_date'])
+        e2_pub = int(e2['publish_date'])
+        if within(e1_pub, e2_pub, 2):
+            return ('date', '+/-2 years', -25)
+        else:
+            return ('date', 'mismatch', -250)
+    except ValueError as TypeError:
+        return ('date', 'mismatch', -250)
+
+
+def compare_isbn10(e1: dict, e2: dict):
```

Comment:
Despite being named `isbn10` this looks like it is comparing any and all ISBNs, but without normalization?

The return value `isbn_match` is set to 85 elsewere in this file, and is effectively a constant.